experiments/sharedUtils/BoxGenerator.js

```diff
@@ -1,127 +1,157 @@
     }
 
 }
 
 //stores the configurations that will control (turn on) lights
-function Config(arg){
+function Config(arg, numButtons){
     this.controls = arg
-}
-
-// stores the current button settings
-function Setting(arg){ // buttons currently pressed
-    this.args = arg
+    this.numButtons = numButtons
 }
 
 //generate the box. Takes in a config (what turns on what) and returns a function that
 // determine what is turned on
 function generateBox(config){
     //function to return, which will take in a setting and return what is turned on
+    function configDict(){
+        let dict = {} // keys: setting of buttons, values: lights
+        for(var i = 0; i < (Math.pow(2, config.numButtons)); i++){
+            var set = ""
+            var setting = {}
+            for(var j = 1; j <= config.numButtons; j++){
+                if((i%(Math.pow(2,j)) >= (Math.pow(2, j-1)))){
+                    set = set + "1"
+                }
+                else {
+                    set = set + "0"
+                }
+                setting[`button${j}`] = (i%(Math.pow(2,j)) >= (Math.pow(2, j-1)))
+            }
+            dict[set] = box(setting)
+        }
+        return dict
+    }
     function box(setting){    
         let arr = [] //array of results
-        for(let i = 0; i < config.controls.length; i++){
-            arr.push(config.controls[i].eval(setting.args))
+        for(var i = 0; i < config.controls.length; i++){
+            arr.push(config.controls[i].eval(setting))
         }
         return arr //return the array
     }
-    return box //return the function
+    return configDict //return the function
 }
 
 // Takes in a configuration, and returns a string representation
 function toString(config){
     var dict = {}
-    for(let i = 0; i < config.controls.length; i++){
+    for(var i = 0; i < config.controls.length; i++){
         dict[`Light ${i+1}`] = {
-            type: config.controls[i].type
-        }
-        for(let j = 0; j < config.controls[i].arg.length; j++){
-            dict[`Light ${i+1}`][`b_${j+1}`] = config.controls[i].arg[j]
+            type: config.controls[i].type,
+            controls: config.controls[i].arg,
         }
     }
     return JSON.stringify(dict)
 }
 
-function createRandomBox(numButtons, numLights){
+function createRandomBox(numButtons, numLights, numStars){
     let configArray = []
     // create a control for each light
     for(let i = 0; i < numLights; i++){
-        console.log("creating a control for light: " + i)
-        let buttonArray = []
-        for(let j = 0; j < numButtons; j++){
-            buttonArray.push(Math.floor(Math.random()*2) === 1)
-        }
+        let compsArray = [] //array of components that control the light
+        let compsDict = {} //dict of components and their settings
+        do{
+            compsArray = []
+            compsDict = {}
+            for(let j = 0; j < numButtons; j++){
+                if(Math.floor(Math.random()*2) === 1){
```

Comment:
As we discussed, prefer the simpler `Math.random() < 0.5` (or any configurable percentage)

I don't think we need to configure the percentage yet, but if we do, consider making it a property of the game, or just an all-caps constant somewhere